src/lakefs_spec/spec.py

```diff
@@ -398,6 +458,7 @@ def __init__(
         fs,
         path,
         mode="rb",
+        source_branch="main",
```

Comment:
This is not needed in the file constructor, since you can get it out of the file system as `self.fs.source_branch`.